### PR TITLE
Cross-platform and project integration fixes

### PR DESCRIPTION
**Pull request by petewarden**

### CODE REVIEW

File: BUILD [hunk 1]
```diff
@@ -67,6 +67,8 @@ cc_library(
     hdrs = [
         ":gemmlowp_public_headers",
     ],
+    includes = ["."],
+    linkopts = ["-pthread"],
```
Comment (bjacob) [inline BUILD]: -pthread and not -lpthread?  Is there a special-casing of pthread by the linker here?

Comment (petewarden) [inline BUILD]: It is a bit funky I know! I actually based this off the Bazel documentation here:
http://bazel.io/docs/cpp.html
From digging into it a bit more, it seems like it does a few other things besides linking in pthreads, like making errno (ick) thread-local:
http://stackoverflow.com/questions/2127797/gcc-significance-of-pthread-flag-when-compiling

Comment (petewarden) [inline BUILD]: It is a bit funky I know! I actually based this off the Bazel documentation here:
http://bazel.io/docs/cpp.html
From digging into it a bit more, it seems like it does a few other things besides linking in pthreads, like making errno (ick) thread-local:
http://stackoverflow.com/questions/2127797/gcc-significance-of-pthread-flag-when-compiling

Comment (petewarden) [inline BUILD]: It is a gcc-ism, but one that a lot of projects seem to suffer from. It's ignored with a warning on OS X.

**Comment by petewarden** [comment]
I switched to the more standard -lpthread way of pulling in the library, and checked that both the Ubuntu and OS X builds still work.

**Comment by bjacob** [comment]
Many thanks!

**Pull request merged**